postgis/postgis.sql.in

```diff
@@ -2859,6 +2859,24 @@ CREATE OR REPLACE FUNCTION ST_Transform(geom geometry, from_proj text, to_srid i
 	LANGUAGE 'sql' IMMUTABLE STRICT PARALLEL SAFE
 	_COST_HIGH;
 
+CREATE OR REPLACE FUNCTION postgis_spatial_ref_sys(auth_name text)
+	RETURNS SETOF TEXT
+	AS 'MODULE_PATHNAME', 'postgis_srs_codes'
+	LANGUAGE 'c' IMMUTABLE STRICT PARALLEL SAFE
+	_COST_MEDIUM;
+
+CREATE OR REPLACE FUNCTION postgis_spatial_ref_sys(auth_name text, auth_srid text)
```

Comment:
maybe this could be a function with all arguments being optional ? You could then only specify an auth_name or both auth_name and auth_srid or nothing (to get a full dump)

Reply:
It feels kind of icky to return the four-entry tuple with only two entries filled in. Returning the setof text made more sense? Another completely different approach would be a auth_names() function that returned setof text and an auth_codes(text) function that returned a setof text and then the ref_sys(text, text) function that returns the actual tuple for that pairing, and then users can compose them to generate whatever complete set of tuples they are looking for.

Reply:
Calling two functions sounds icky.  Honestly I don't see why I would use these functions to do anything but populate spatial_ref_sys so it's unclear to me why I would want to call two functions to do that.  The proliferation of functions is also icky - way more icky than returning NULLs.

I very well could be missing something.  Would be way easier if you just commit stuff and we can work out the kinks later.  I hate following multiple branches to see what's up.